plugins/modules/ec2_scaling_policy.py

```diff
@@ -91,97 +93,108 @@ def create_scaling_policy(connection, module):
+                    AdjustmentType=adjustment_type,
+                    AutoScalingGroupName=asg_name,
+                    ScalingAdjustment=scaling_adjustment,
+                    MinAdjustmentStep=min_adjustment_step,
+                    Cooldown=cooldown)
+                policy = connection.describe_policies(AutoScalingGroupName=asg_name,
+                                                      PolicyNames=[sp_name])['ScalingPolicies'][0]
+                module.exit_json(changed=True, name=policy['PolicyName'], arn=policy['PolicyARN'],
+                                 as_name=policy['AutoScalingGroupName'], scaling_adjustment=policy['ScalingAdjustment'],
+                                 cooldown=policy['Cooldown'], adjustment_type=policy['AdjustmentType'],
+                                 min_adjustment_step=policy['MinAdjustmentStep'])
+            except ClientError as e:
+                module.fail_json_aws(e)
 
-        try:
-            connection.create_scaling_policy(sp)
-            policy = connection.get_all_policies(as_group=asg_name, policy_names=[sp_name])[0]
-            module.exit_json(changed=True, name=policy.name, arn=policy.policy_arn, as_name=policy.as_name, scaling_adjustment=policy.scaling_adjustment,
-                             cooldown=policy.cooldown, adjustment_type=policy.adjustment_type, min_adjustment_step=policy.min_adjustment_step)
-        except BotoServerError as e:
-            module.fail_json(msg=str(e))
     else:
-        policy = scalingPolicies[0]
+        policy = scaling_policies['ScalingPolicies'][0]
         changed = False
 
         # min_adjustment_step attribute is only relevant if the adjustment_type
         # is set to percentage change in capacity, so it is a special case
-        if getattr(policy, 'adjustment_type') == 'PercentChangeInCapacity':
-            if getattr(policy, 'min_adjustment_step') != module.params.get('min_adjustment_step'):
+        if policy.get('AdjustmentType') == 'PercentChangeInCapacity':
+            if policy.get('MinAdjustmentStep') != min_adjustment_step:
                 changed = True
 
-        # set the min adjustment step in case the user decided to change their
-        # adjustment type to percentage
-        setattr(policy, 'min_adjustment_step', module.params.get('min_adjustment_step'))
+            # set the min adjustment step in case the user decided to change their
+            # adjustment type to percentage
+            policy['MinAdjustmentStep'] = min_adjustment_step
 
         # check the remaining attributes
-        for attr in ('adjustment_type', 'scaling_adjustment', 'cooldown'):
-            if getattr(policy, attr) != module.params.get(attr):
+        for attr in ('AdjustmentType', 'ScalingAdjustment', 'Cooldown'):
+            if policy.get(attr) != module.params.get(attr):
                 changed = True
-                setattr(policy, attr, module.params.get(attr))
+                policy[attr] = module.params.get(attr)
```

Comment:
This looks wrong; module parameters are snake_case, but attr is now CamelCase.

Reply:
It looks like I might be able to use something like snake_dict_to_camel_dict as seen here https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/ec2_lc.py#L541. I'll give that a try.

Reply:
I believe I have a change for this, once I'm able to test it I can commit it to this branch for review.